storage/filesystem/object.go

```diff
@@ -71,7 +71,14 @@ func (s *ObjectStorage) Get(h core.Hash) (core.Object, error) {
 	p := packfile.NewParser(r)
 
 	obj := s.NewObject()
-	return obj, p.FillObject(obj)
+	err = p.FillObject(obj)
+	if err != nil {
+		return nil, err
+	}
+	if core.AnyObject != t && obj.Type() != t {
```

Comment:
I think this check should be made before the FillObject operation, it is the less surprise path as it would be the logical order to avoid doing unnecessary extra work in the case of a wrong obj.Type.